elide-core/src/main/java/com/yahoo/elide/core/EntityDictionary.java

```diff
@@ -1214,6 +1214,35 @@ public boolean hasBinding(Class<?> cls) {
         return bindJsonApiToEntity.contains(cls);
     }
 
+    /**
+     * Returns whether or not a specified annotation is present on an entity field or its corresponding method.
+     *
+     * @param fieldName  The entity field
+     * @param annotationClass  The provided annotation class
+     *
+     * @param <A>  The type of the {@code annotationClass}
+     *
+     * @return {@code true} if the field is annotated by the {@code annotationClass}
+     */
+    public <A extends Annotation> boolean attributeOrRelationAnnotationExists(
+            Class<?> cls,
+            String fieldName,
+            Class<A> annotationClass
+    ) {
+        ConcurrentHashMap<String, AccessibleObject> fieldOrMethods = getEntityBinding(cls).fieldsToValues;
+        if (fieldOrMethods == null) {
+            return false;
+        }
+
+        AccessibleObject fieldOrMethod = fieldOrMethods.get(fieldName);
+        if (fieldOrMethod == null) {
+            return false;
+        }
+
+        return getAttributeOrRelationAnnotation(cls, annotationClass, EntityBinding.getFieldName(fieldOrMethod))
```

Comment:
This method can be collapsed to just:
```java
return getAttributeOrRelationAnnotation(cls, annotationClass, EntityBinding.getFieldName(fieldOrMethod)) != null
```

Reply:
Right. To get `fieldOrMethod` used in this line, we probably need to execute extra logic above.